src/azure-cli/azure/cli/command_modules/cosmosdb/custom.py

```diff
@@ -1426,39 +1446,112 @@ def cli_cosmosdb_identity_show(client, resource_group_name, account_name):
                                  resource_group_name,
-                                 account_name):
-    """ Remove the SystemAssigned identity associated with a Cosmos DB account """
+                                 account_name,
+                                 identities=None):
+    """ Remove the identities associated with a Cosmos DB account """
 
     existing = client.get(resource_group_name, account_name)
 
-    if ResourceIdentityType.system_assigned.value not in existing.identity.type:
-        return existing.identity
-
-    if ResourceIdentityType.user_assigned.value in existing.identity.type:
-        identity = ManagedServiceIdentity(type=ResourceIdentityType.user_assigned.value)
+    SYSTEM_ID = '[system]'
+    remove_system_assigned_identity = False
+    if not identities:
+        remove_system_assigned_identity = True
+    elif SYSTEM_ID in identities:
+        remove_system_assigned_identity = True
+        identities.remove(SYSTEM_ID)
+
+    if existing.identity is None:
+        return ManagedServiceIdentity(type=ResourceIdentityType.none.value)
+    if existing.identity.user_assigned_identities:
+        existing_identities = existing.identity.user_assigned_identities.keys()
     else:
-        identity = ManagedServiceIdentity(type=ResourceIdentityType.none.value)
-    params = DatabaseAccountUpdateParameters(identity=identity)
+        existing_identities = []
+    if identities:
+        identities_to_remove = identities
+    else:
+        identities_to_remove = []
+    non_existing = [x for x in identities_to_remove if x not in set(existing_identities)]
+
+    if non_existing:
+        raise CLIError("'{}' are not associated with '{}'".format(','.join(non_existing), account_name))
+    identities_remaining = [x for x in existing_identities if x not in set(identities_to_remove)]
+    if remove_system_assigned_identity and ((not existing.identity) or (existing.identity and existing.identity.type in [ResourceIdentityType.none, ResourceIdentityType.user_assigned])):
+        raise CLIError("System-assigned identity is not associated with '{}'".format(account_name))
+
+    if identities_remaining and not remove_system_assigned_identity and existing.identity.type == ResourceIdentityType.system_assigned_user_assigned:
+        set_type = ResourceIdentityType.system_assigned_user_assigned
+    elif identities_remaining and remove_system_assigned_identity and existing.identity.type == ResourceIdentityType.system_assigned_user_assigned:
+        set_type = ResourceIdentityType.user_assigned
+    elif identities_remaining and not remove_system_assigned_identity and existing.identity.type == ResourceIdentityType.user_assigned:
+        set_type = ResourceIdentityType.user_assigned
+    elif not identities_remaining and not remove_system_assigned_identity and existing.identity.type == ResourceIdentityType.system_assigned_user_assigned:
+        set_type = ResourceIdentityType.system_assigned
```

Comment:
Also curious here why you're using the "set_type"? I mean the prefix "set_" looks odd to me. Generally we use new_type or other naming